clientapi/routing/login.go

```diff
@@ -40,28 +41,24 @@ type flow struct {
 	Type string `json:"type"`
 }
 
-func passwordLogin() flows {
-	f := flows{}
-	s := flow{
-		Type: "m.login.password",
-	}
-	f.Flows = append(f.Flows, s)
-	return f
-}
-
 // Login implements GET and POST /login
 func Login(
 	req *http.Request, userAPI userapi.ClientUserAPI,
 	cfg *config.ClientAPI,
 ) util.JSONResponse {
 	if req.Method == http.MethodGet {
-		// TODO: support other forms of login other than password, depending on config options
+		// TODO: support other forms of login, depending on config options
 		return util.JSONResponse{
 			Code: http.StatusOK,
-			JSON: passwordLogin(),
+			JSON: flows{
+				Flows: []flow{
+					{Type: authtypes.LoginTypePassword},
+					{Type: authtypes.LoginTypeApplicationService},
```

Comment:
As in the other PR, we should only add `authtypes.LoginTypeApplicationService` if we have application services enabled. e.g. `if len(cfg.Derived.ApplicationServices) > 0 {}`

Reply:
I do get that sentiment - why expose a login method if it isn't available - reduces the amount of attack vectors bad actors could invoke on a server. Added a commit for that.